File: mini_gplus/models/notification.py
from enum import Enum
from mongoengine import Document, LazyReferenceField, StringField, EnumField, BooleanField, CASCADE
from .created_at_mixin import CreatedAtMixin
from .user import User


class NotifyingAction(Enum):
    Comment = "comment"
    Mention = "mention"
    Reaction = "reaction"
    Reshare = "reshare"
    Follow = "follow"


class Notification(Document, CreatedAtMixin):
    eid = StringField(required=True)
    notifying_action = EnumField(NotifyingAction, required=True)  # type: NotifyingAction
    unread = BooleanField(required=False, default=True)

    notifier = LazyReferenceField(User, required=True, reverse_delete_rule=CASCADE)  # type: User
    notifying_href = StringField(required=True)
    notifying_summary = StringField(required=False, default='')
    notifying_deleted = BooleanField(required=False, default=False)

    owner = LazyReferenceField(User, required=True, reverse_delete_rule=CASCADE)  # type: User
    notified_href = StringField(required=True)
    notified_summary = StringField(required=False, default='')
    notified_deleted = BooleanField(required=False, default=False)

File: web/src/models/Notification.ts
import User from "./User";

export type NotifyingAction = 'comment' | 'mention' | 'reaction' | 'reshare' | 'follow'

export default interface Notification {
  notified_summary: string;
  created_at_seconds: number;
  id: string;
  notified_href: string;
  notified_deleted: boolean;
  notifier: User;
  notifier_blocked: boolean;
  notifying_deleted: boolean;
  notifying_summary: string;
  unread: boolean;
  notifying_action: NotifyingAction;
}
